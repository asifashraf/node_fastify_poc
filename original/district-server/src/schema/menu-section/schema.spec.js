const { fetchGraphQL, getFirstId, testDb } = require('../../lib/test-util');
const {
  menuSectionDetails,
  sampleBrandId,
} = require('../../lib/test-fragments');
const { map, extend } = require('lodash');

test('menus can resolve menu sections', async () => {
  const query = `{
    brand(id: "${sampleBrandId}") {
      menu(countryId:"01c73b60-2c6a-45f1-8dbf-88a5ce4ad179") {
        sections {
          ${menuSectionDetails}
        }
      }
    }
  }`;
  // openInGraphiql(query);
  const result = await fetchGraphQL(query);
  expect(result).toMatchSnapshot();
});

test('menu items can resolve their menu sections', async () => {
  const query = `{
    brand(id: "${sampleBrandId}") {
      menu(countryId:"01c73b60-2c6a-45f1-8dbf-88a5ce4ad179") {
        sections {
          items {
            section {
              ${menuSectionDetails}
            }
          }
        }
      }
    }
  }`;
  // openInGraphiql(query);
  const result = await fetchGraphQL(query);
  expect(result).toMatchSnapshot();
});

test('menu sections can be saved', async () => {
  const { id: menuId } = await getFirstId('menus');

  const query = `
    mutation {
      brandMenuSectionSave(menuSections:
      [
        { name: {
          en: "Le Menu Section 1 "
          ar: "لو القائمة القسم 1"
        }, menuId:"${menuId}" }
        { name: {
          en: "Le Menu Section 2 "
          ar: "لو القائمة القسم 2"
        }, menuId:"${menuId}" }
        { name: {
          en: "Le Menu Section 3 "
          ar: "لو القائمة القسم 3"
        }, menuId:"${menuId}" }
       ])
      {
        error
        errors
        menuSections {
          id
          name {
            en
            ar
          }
        }
      }
    }`;
  // openInGraphiql(query);
  const result = await fetchGraphQL(query);
  expect(result).toMatchSnapshot();
});

test('menu sections can be updated', async () => {
  const { id: menuId } = await getFirstId('menus');

  const sections = map(
    await testDb
      .handle('menu_sections')
      .select('id')
      .where('menu_id', menuId),
    (item, index) => {
      const newItem = extend({}, item, {
        name: {
          en: `Le Menu Section ${index}`,
          ar: `لو القائمة القسم ${index}`,
        },
      });
      return `{ id: "${newItem.id}", name: { en: "${newItem.name.en}", ar: "${newItem.name.ar}"   }, menuId:"${menuId}"}`;
    }
  );

  const query = `
    mutation {
      brandMenuSectionSave(menuSections: [ ${sections} ]
      )
      {
        error
        errors
        menuSections {
          id
          name {
            en
            ar
          }
        }
      }
    }`;
  // openInGraphiql(query);
  const result = await fetchGraphQL(query);
  expect(result).toMatchSnapshot();
});
