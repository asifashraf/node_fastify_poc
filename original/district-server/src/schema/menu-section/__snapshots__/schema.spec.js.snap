// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`menu items can resolve their menu sections 1`] = `
Object {
  "data": Object {
    "brand": Object {
      "menu": Object {
        "sections": Array [
          Object {
            "items": Array [
              Object {
                "section": Object {
                  "id": "62ee1779-4312-40ee-95c4-7cce956c11ae",
                  "name": Object {
                    "ar": "مشروبات",
                    "en": "Drinks",
                  },
                },
              },
              Object {
                "section": Object {
                  "id": "62ee1779-4312-40ee-95c4-7cce956c11ae",
                  "name": Object {
                    "ar": "مشروبات",
                    "en": "Drinks",
                  },
                },
              },
              Object {
                "section": Object {
                  "id": "62ee1779-4312-40ee-95c4-7cce956c11ae",
                  "name": Object {
                    "ar": "مشروبات",
                    "en": "Drinks",
                  },
                },
              },
            ],
          },
          Object {
            "items": Array [
              Object {
                "section": Object {
                  "id": "ac7a3fe1-5d8d-4f74-984c-eecb7c105adb",
                  "name": Object {
                    "ar": "طعام",
                    "en": "Food",
                  },
                },
              },
              Object {
                "section": Object {
                  "id": "ac7a3fe1-5d8d-4f74-984c-eecb7c105adb",
                  "name": Object {
                    "ar": "طعام",
                    "en": "Food",
                  },
                },
              },
              Object {
                "section": Object {
                  "id": "ac7a3fe1-5d8d-4f74-984c-eecb7c105adb",
                  "name": Object {
                    "ar": "طعام",
                    "en": "Food",
                  },
                },
              },
            ],
          },
          Object {
            "items": Array [
              Object {
                "section": Object {
                  "id": "6c52f1d1-7d39-41bb-8b0f-6e147183265c",
                  "name": Object {
                    "ar": "مجمد",
                    "en": "Frozen",
                  },
                },
              },
              Object {
                "section": Object {
                  "id": "6c52f1d1-7d39-41bb-8b0f-6e147183265c",
                  "name": Object {
                    "ar": "مجمد",
                    "en": "Frozen",
                  },
                },
              },
            ],
          },
        ],
      },
    },
  },
}
`;

exports[`menu sections can be saved 1`] = `
Object {
  "data": Object {
    "brandMenuSectionSave": Object {
      "error": null,
      "errors": null,
      "menuSections": Array [
        Object {
          "id": "62ee1779-4312-40ee-95c4-7cce956c11ae",
          "name": Object {
            "ar": "مشروبات",
            "en": "Drinks",
          },
        },
        Object {
          "id": "9f03ad20-3fc0-4896-8917-35324e810d58",
          "name": Object {
            "ar": "لو القائمة القسم 1",
            "en": "Le Menu Section 1 ",
          },
        },
        Object {
          "id": "ac7a3fe1-5d8d-4f74-984c-eecb7c105adb",
          "name": Object {
            "ar": "طعام",
            "en": "Food",
          },
        },
        Object {
          "id": "c81d4284-9dbf-4dee-9e71-72d5546407a7",
          "name": Object {
            "ar": "لو القائمة القسم 2",
            "en": "Le Menu Section 2 ",
          },
        },
        Object {
          "id": "6c52f1d1-7d39-41bb-8b0f-6e147183265c",
          "name": Object {
            "ar": "مجمد",
            "en": "Frozen",
          },
        },
        Object {
          "id": "1e39e901-0b91-4ff9-b741-98042a8f1a14",
          "name": Object {
            "ar": "لو القائمة القسم 3",
            "en": "Le Menu Section 3 ",
          },
        },
      ],
    },
  },
}
`;

exports[`menu sections can be updated 1`] = `
Object {
  "data": Object {
    "brandMenuSectionSave": Object {
      "error": null,
      "errors": null,
      "menuSections": Array [
        Object {
          "id": "62ee1779-4312-40ee-95c4-7cce956c11ae",
          "name": Object {
            "ar": "لو القائمة القسم 0",
            "en": "Le Menu Section 0",
          },
        },
        Object {
          "id": "ac7a3fe1-5d8d-4f74-984c-eecb7c105adb",
          "name": Object {
            "ar": "لو القائمة القسم 1",
            "en": "Le Menu Section 1",
          },
        },
        Object {
          "id": "6c52f1d1-7d39-41bb-8b0f-6e147183265c",
          "name": Object {
            "ar": "لو القائمة القسم 2",
            "en": "Le Menu Section 2",
          },
        },
      ],
    },
  },
}
`;

exports[`menus can resolve menu sections 1`] = `
Object {
  "data": Object {
    "brand": Object {
      "menu": Object {
        "sections": Array [
          Object {
            "id": "62ee1779-4312-40ee-95c4-7cce956c11ae",
            "name": Object {
              "ar": "مشروبات",
              "en": "Drinks",
            },
          },
          Object {
            "id": "ac7a3fe1-5d8d-4f74-984c-eecb7c105adb",
            "name": Object {
              "ar": "طعام",
              "en": "Food",
            },
          },
          Object {
            "id": "6c52f1d1-7d39-41bb-8b0f-6e147183265c",
            "name": Object {
              "ar": "مجمد",
              "en": "Frozen",
            },
          },
        ],
      },
    },
  },
}
`;
